### PR TITLE
feat: add coverage

### PR DESCRIPTION
**Pull request by anhnd350309**
### Description
This PR introduces code coverage tracking to the project. Key additions include:
- check coverage in CI-CD: example 
```bash
Total Coverage: 25.05%
Coverage (25.05%) meets the threshold (20%).
```
- If the coverage is less than 20%, the PR cannot be merged.
- If you want to run in locally, run `pnpm cd-test` and check the `coverage` folder
### Next Steps
- Review and validate coverage thresholds
- Ensure CI/CD pipeline integration for automated coverage checks.

### CODE REVIEW

**Review by JanLewDev — APPROVED** [review]
lgtm

**Review by trungnotchung — APPROVED** [review]
lgtm

File: check-coverage.ts
Comment (d-roak) [inline check-coverage.ts:1]: Add this file to 'scripts/'

File: package.json [hunk 1]
```diff
@@ -28,12 +28,14 @@
 		"release": "release-it",
 		"test": "vitest",
 		"ci-test": "vitest --exclude **/node.test.ts",
-		"watch": "pnpm --parallel --filter '@ts-drp/*' watch"
+		"watch": "pnpm --parallel --filter '@ts-drp/*' watch",
+		"check-coverage": "npx ts-node check-coverage.ts"
```
Comment (d-roak) [inline package.json:32]: Change the name to 'coverage', and put it in alphabetical order with the other commands

File: vite.config.mts
Comment (d-roak) [inline vite.config.mts:1]: Remove comments

File: package.json (hunk 1 above)
Comment (d-roak) [reply package.json:32]: 'pnpm exec' instead of 'npx'

File: check-coverage.ts
Comment (anhnd350309) [reply check-coverage.ts:1]: Done

File: package.json (hunk 1 above)
Comment (anhnd350309) [reply package.json]: done and tested

File: vite.config.mts
Comment (anhnd350309) [reply vite.config.mts:1]: done

**Review by d-roak — APPROVED** [review]
lgtm

**Pull request merged**